### PR TITLE
Fix for DOI copy issues

### LINKED ISSUE #8389: 'Copy DOI URL' in entry list only copies DOI but not the URL
### JabRef version

Latest development branch build (please note build date below)
JabRef 5.5--2022-01-03--c94bb1b
Linux 5.11.0-44-generic amd64 
Java 16.0.2 
JavaFX 17.0.1+1

### Operating system

GNU / Linux

### Details on version and operating system

Ubuntu 20.04

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. In the entry list, right-click on a URL symbol in the URL column and chose menu item "Copy ... -> Copy DOI URL"
2. Paste from clipboard and you will only see the DOI pasted. 
3. Now, open that entry in the entry editor, go to "Optional fields", right-click on DOI and chose menu item "Copy DOI URL".
4. Paste from clipboard and you will see the DOI URL pasted.

### PR DESCRIPTION
**Pull request by Shinigami**
fixes #8389 

Modified the StandardAction `COPY_DOI` from `COPY_DOI(Localization.lang("Copy DOI url"))` to `COPY_DOI(Localization.lang("Copy DOI"))`. This is because COPY_DOI relied on a function that only returned the DOI rather than the DOI url. This was changed to be more accurate.

Added the StandardAction `COPY_DOI_URL(Localization.lang("Copy DOI url"))`. In addition, added `case: (COPY_DOI_URL -> copyDoiUrl())` to CopyMoreAction.java. Along with this, I added a new function `private void copyDoiUrl()` which does the same as `copyDoi()` except it uses `entry.getDOI().get().getURIAsASCIIString()` to collect all DOI urls rather than just the DOI. This is because there was a conflict between the test cases in CopyMoreActionTest.java and COPY_DOI. The former expected just the DOI while the latter expected a DOI url.  So, when copying the DOI url in the entry list, it would cause the test cases  `testExecuteCopyDoiOnPartialSuccess` and `testExecuteCopyDoiOnSuccess` in CopyMoreActionTest.java to fail. To remove this conflict, accomodate the test cases, and prevent any misconceptions, I added a separate action solely for copying the DOI url in addition to the action to copy the DOI. This also enables any future use to request either option.

Modified `copyDoiUrlMenuItem` in EditorMenus.java to match proper action `COPY_DOI_URL`. Furthermore, I added a new MenuItem `factory.createMenuItem(StandardActions.COPY_DOI_URL, new CopyMoreAction(StandardActions.COPY_DOI_URL, ...)` in RightClickMenu.java to differentiate between "Copy DOI" and "Copy DOI url". This is because `copyDoi()` was only utilized by the test cases mentioned above, and I did not see an option to just copy the DOI in the entry list. Since the entry editor lets you copy both, and to prevent redundancy, I let the entry list do the same for consistency. 

The screenshots for the UI change are shown below.

Before:
image

After:
image

Added `Copy\ DOI=Copy DOI` to JabRef_en.properties as a result of adding the new Localization.lang entry `"Copy DOI"`.

Finally, added changes to CHANGELOG.md. 

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/edit/CopyMoreAction.java [hunk 1]
```diff
@@ -145,6 +146,33 @@ private void copyDoi() {
...
+    private void copyDoiUrl() {
+        List<BibEntry> entries = stateManager.getSelectedEntries();
+
+        // Collect all non-null DOI urls.
+        List<String> dois = entries.stream()
+                .filter(entry -> entry.getDOI().isPresent())
+                .map(entry -> entry.getDOI().get().getURIAsASCIIString())
+                .collect(Collectors.toList());
+
+        if (dois.isEmpty()) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/edit/CopyMoreAction.java:158]: In general looks good so far but now you have a lot of code duplication with the other method.
So why not create a common method where you just have the list of dois/doi urls as input parameter?

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
It would be better if you extract the duplicated code in a common method and just pass the dois in as parameter

File: src/main/java/org/jabref/gui/edit/CopyMoreAction.java (hunk 1 above)
Comment (Shinigami) [reply src/main/java/org/jabref/gui/edit/CopyMoreAction.java:158]: I overlooked this, thank you for the feedback. I will change this.

**Review by Siedlerchr — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]
Thanks for the follow up!

**Review by koppor — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Thanks a lot for your contribution! Really useful!

**Pull request merged**